go/libkb/user.go

```diff
@@ -917,6 +922,10 @@ func LoadHasRandomPw(mctx MetaContext, arg keybase1.LoadHasRandomPwArg) (res boo
 		return res, err
 	}
 
+	if arg.ForUID != nil && !arg.ForUID.Equal(currentUID) {
+		return res, errors.New("current user changed")
```

Comment:
maybe all this is not needed anymore if the context is cancelled after logout

Reply:
Why even do this comparison? Just use the UID in the param.

Reply:
the idea was that the uid might change while the https request is still in flight. but maybe this is not needed since we are cancelling the context. not sure if onLogout fires during user switching though, i need to find out

Reply:
@mmaxim so after remove all that `ForUID` stuff, will it be good to go?

Reply:
I think you can also simplify the logout hook stuff per my other comment.